src/main/java/org/sonarlint/intellij/config/project/AddEditExclusionDialog.java

```diff
@@ -0,0 +1,232 @@
+    globTextField.setEnabled(globRadioButton.isSelected());
+    fileTextField.setEnabled(fileRadioButton.isSelected());
+    helpLabel.setEnabled(globRadioButton.isSelected());
+
+    updateOk();
+  }
+
+  private void updateOk() {
+    boolean valid = StringUtils.trimToNull(getSelectedText().getText()) != null;
+    myOKAction.setEnabled(valid);
+  }
+
+  private JTextField getSelectedText() {
+    if (directoryRadioButton.isSelected()) {
+      return directoryTextField.getTextField();
+    } else if (fileRadioButton.isSelected()) {
+      return fileTextField.getTextField();
+    } else {
+      return globTextField;
+    }
+  }
+
+  @Nullable
+  protected JComponent createCenterPanel() {
+    return panel;
+  }
+
+  @CheckForNull
+  public ExclusionItem getExclusion() {
+    if (directoryRadioButton.isSelected()) {
+      return new ExclusionItem(DIRECTORY, relative(directoryTextField.getText()));
+    } else if (fileRadioButton.isSelected()) {
+      return new ExclusionItem(FILE, relative(fileTextField.getText()));
+    } else {
+      return new ExclusionItem(GLOB, relative(globTextField.getText()));
+    }
+  }
+
+  public void setExclusion(@Nullable ExclusionItem item) {
+    setTitle("Edit SonarLint File Exclusion");
+
+    if (item != null) {
+      switch (item.type()) {
+        case DIRECTORY:
+          directoryTextField.setText(item.item());
+          directoryRadioButton.setSelected(true);
+          break;
+        case FILE:
+          fileTextField.setText(item.item());
+          fileRadioButton.setSelected(true);
+          break;
+        case GLOB:
+          globTextField.setText(item.item());
+          globRadioButton.setSelected(true);
+          break;
+      }
+      updateControls();
+    }
+  }
+
+  private void createUIComponents() {
+    patternHelp = new HyperlinkLabel("");
+    Icon infoIcon = SonarLintIcons.INFO;
+    patternHelp.setIcon(infoIcon);
+    //hoverHyperlinkLabel1.setUseIconAsLink(true);
+    patternHelp.addMouseListener(new MouseAdapter() {
+      public void mouseEntered(MouseEvent e) {
+        final JLabel label = new JLabel("<<html><body style=\"margin-bottom:1pt\">Configure GLOB patterns to be applied to files in all projects.<br/><br/><b>Examples:</b><br/><table border=\"1\" cellpadding=\"1\" cellspacing=\"0\"><tr><td><pre>**/*.js</pre></td><td>Exclude all javascript files</td></tr><tr><td><pre>src/main/test/**</pre></td><td>Exclude all test sources</td></tr><tr><td><pre>**/*{!.java}</pre></td><td>Exclude everything except Java files</td></tr></table></body></html>");
+        label.setBorder(HintUtil.createHintBorder());
+        label.setBackground(HintUtil.INFORMATION_COLOR);
+        label.setOpaque(true);
+        HintManager.getInstance().showHint(label, RelativePoint.getSouthWestOf(patternHelp),
+          HintManager.HIDE_BY_ANY_KEY | HintManager.HIDE_BY_TEXT_CHANGE, -1);
+      }
+
+      public void mouseExited(MouseEvent e) {
+       HintManager.getInstance().hideAllHints();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this line start at column 9. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)